block-producer/src/state_view/mod.rs

```diff
@@ -179,7 +179,9 @@ fn ensure_tx_inputs_constraints(
 
     match tx_inputs.account_hash {
         Some(store_account_hash) => {
-            if candidate_tx.initial_account_hash() != store_account_hash {
+            if candidate_tx.initial_account_hash() != store_account_hash
+                && store_account_hash != Digest::default()
+            {
```

Comment:
I would document this much better and would probably split into two branches for clarity (even if ` store_account_hash != Digest::default()` branch is a no-op)

Reply:
This PR should have been in draft. 

Absolutely, added your demands.